prime-router/src/test/kotlin/common/UniversalPipelineTestUtils.kt

```diff
@@ -43,14 +43,27 @@ const val validFHIRRecord1 =
 const val validFHIRRecord1Identifier = "1234d1d1-95fe-462c-8ac6-46728dba581c"
 
+@Suppress("ktlint:standard:max-line-length")
+const val validFHIRRecord1a =
+    """{"resourceType":"Bundle","id":"1667861767830636000.7db38d22-b713-49fc-abfa-2edba9c12347","meta":{"lastUpdated":"2022-11-07T22:56:07.832+00:00"},"identifier":{"value":"1234d1d1-95fe-462c-8ac6-46728dba581c"},"type":"message","timestamp":"2021-08-03T13:15:11.015+00:00","entry":[{"fullUrl":"MessageHeader/0993dd0b-6ce5-3caf-a177-0b81cc780c18","resource":{"resourceType":"MessageHeader","id":"0993dd0b-6ce5-3caf-a177-0b81cc780c18","extension":[{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/encoding-characters","valueString":"^~\\&#"},{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/character-set","valueString":"UNICODE UTF-8"},{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/msh-message-header","extension":[{"url":"MSH.7","valueString":"20230501102531-0400"}]}],"eventCoding":{"system":"http://terminology.hl7.org/CodeSystem/v2-0003","code":"R01","display":"ORU^R01^ORU_R01"},"sender":{"reference":"Organization/1710886092467181000.213628f7-9569-4400-a95d-621c3bfbf121"}}},{"fullUrl":"Observation/d683b42a-bf50-45e8-9fce-6c0531994f09","resource":{"resourceType":"Observation","id":"d683b42a-bf50-45e8-9fce-6c0531994f09","status":"final","code":{"coding":[{"system":"http://loinc.org","code":"80382-5"}],"text":"Flu A"},"subject":{"reference":"Patient/9473889b-b2b9-45ac-a8d8-191f27132912"},"performer":[{"reference":"Organization/1a0139b9-fc23-450b-9b6c-cd081e5cea9d"}],"valueCodeableConcept":{"coding":[{"system":"http://snomed.info/sct","code":"260373001","display":"Detected"}]},"interpretation":[{"coding":[{"system":"http://terminology.hl7.org/CodeSystem/v2-0078","code":"A","display":"Abnormal"}]}],"method":{"extension":[{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/testkit-name-id","valueCoding":{"code":"BD Veritor System for Rapid Detection of SARS-CoV-2 & Flu A+B_Becton, Dickinson and Company (BD)"}},{"url":"https://reportstream.cdc.gov/fhir/StructureDefinition/equipment-uid","valueCoding":{"code":"BD Veritor System for Rapid Detection of SARS-CoV-2 & Flu A+B_Becton, Dickinson and Company (BD)"}}],"coding":[{"display":"BD Veritor System for Rapid Detection of SARS-CoV-2 & Flu A+B*"}]},"specimen":{"reference":"Specimen/52a582e4-d389-42d0-b738-bee51cf5244d"},"device":{"reference":"Device/78dc4d98-2958-43a3-a445-76ceef8c0698"}}}]}"""
```

Comment:
This was something that I meant to ask before I committed the code.  We never reached a definitive conclusion about whether or not we could expect the event.Coding in the MessageHeader.  If I just add it to the existing variables then it will break other tests.  The 1a versions were added to support my tests without breaking the other existing tests.

Reply:
We should try to not keep duplicating these test messages for specific cases and instead prefer to fix existing tests if it makes to do so.

> We never reached a definitive conclusion about whether or not we could expect the event.Coding in the MessageHeader.

Can you remind me what the issue was here? AFAIK:

- SR is only FHIR sender right now, and they are populating it
- Our HL7v2->FHIR mappings populate it? (did you confirm this?)

Given the above are true, I think we should expect that to be populated in MessageHeader?

Reply:
You are correct in that the example from SimpleReport does contain ORU_R01:

```
{
      "fullUrl": "MessageHeader/a9922147-0ae7-4b2c-bb50-2d373a7efb18",
      "resource": {
        "resourceType": "MessageHeader",
        "id": "a9922147-0ae7-4b2c-bb50-2d373a7efb18",
        "meta": {
          "tag": [
            {
              "system": "http://terminology.hl7.org/CodeSystem/v2-0103",
              "code": "P",
              "display": "Production"
            }
          ]
        },
        "eventCoding": {
          "system": "http://terminology.hl7.org/CodeSystem/v2-0003",
          "code": "ORU_R01",
          "display": "ORU/ACK - Unsolicited transmission of an observation message"
        },
        "destination": [
          {
            "name": "PRIME ReportStream",
            "endpoint": "https://prime.cdc.gov/api/reports?option=SkipInvalidItems"
          }
        ],
        "sender": {
          "reference": "Organization/719ec8ad-cf59-405a-9832-c4065945c130"
        },
        "source": {
          "extension": [
            {
              "url": "https://reportstream.cdc.gov/fhir/StructureDefinition/software-binary-id",
              "valueString": "3fe4aeb"
            },
            {
              "url": "https://reportstream.cdc.gov/fhir/StructureDefinition/software-install-date",
              "valueInstant": "2024-11-14T16:38:42Z"
            },
            {
              "url": "https://reportstream.cdc.gov/fhir/StructureDefinition/software-vendor-org",
              "valueReference": {
                "reference": "Organization/07640c5d-87cd-488b-9343-a226c5166539"
              }
            }
          ],
          "software": "PRIME SimpleReport",
          "version": "3fe4aeb",
          "endpoint": "https://simplereport.gov/"
        },
        "focus": [
          {
            "reference": "Provenance/1810d743-5440-470c-8afc-528b9b35784c"
          },
          {
            "reference": "DiagnosticReport/2b99e03c-b797-497e-8f93-0d71354b22d4"
          }
        ]
      }
    },
```

And our HL7->FHIR mappings do seem to bring forward the HL7 trigger type in the MessageHeader eventCoding.  The issue is, since it is not strictly a required field, can/should we rely on it being populated for all senders?  Do we have the ability to compel them to do so?

Reply:
We do have the ability to compel them to do so (via FHIR profile enforcement, which is not implemented yet) and also during the onboarding process. Do a lot of tests fail if you update the existing bundles in here to include the eventCoding? and why?